client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -21,21 +22,24 @@ const { Option, OptGroup } = Select;
 export class ScheduleDialog extends React.Component {
   static propTypes = {
     show: PropTypes.bool.isRequired,
-    // eslint-disable-next-line react/forbid-prop-types
-    query: PropTypes.object.isRequired,
+    schedule: RefreshScheduleType,
     refreshOptions: PropTypes.arrayOf(PropTypes.number).isRequired,
     updateQuery: PropTypes.func.isRequired,
     onClose: PropTypes.func.isRequired,
   };
 
+  static defaultProps = {
+    schedule: RefreshScheduleDefault,
+  };
+
   constructor(props) {
     super(props);
     this.state = this.initState;
     this.modalRef = React.createRef(); // used by <Select>
   }
 
   get initState() {
-    const newSchedule = clone(this.props.query.schedule);
+    const newSchedule = clone(this.props.schedule || ScheduleDialog.defaultProps.schedule);
```

Comment:
Won't React set the value to the default value on its own, when the passed property is `null`?

Reply:
I WISH! The whole trouble here is that `null` is considered a value (as opposed to `undefined`). WAT?!
Also, `defaultProps` is a shallow nester which means the Object props won't have any meaning.

I do believe this deserves the `typescript-ftw` label here but I need to make sure.

Reply:
🤦‍♂️